src/functions/authoriser/framework/handler.ts

```diff
@@ -1,6 +1,49 @@
-import { CustomAuthorizerEvent, CustomAuthorizerResult, Context } from 'aws-lambda';
+import * as util from 'util';
+import { CustomAuthorizerEvent, CustomAuthorizerResult } from 'aws-lambda';
+import * as jwksClient from 'jwks-rsa';
+import nodeFetch from 'node-fetch';
+import { AdJwtVerifier } from '../application/AdJwtVerifier';
 
-const newCustomAuthorizerResult = (principalId: string, effect: string, resource: string)
+const ensureNotNullOrEmpty = (val: any, fieldName: string) => {
+  if (val === null || val === undefined || val === '') {
+    throw `Error: ${fieldName} is null or empty`;
+  }
+};
+
+/**
+ * Creates an AdJwtVerifier that can verify the authenticity of Azure Active Directory JWTs.
+ */
+const createAdJwtVerifier = async (): Promise<AdJwtVerifier> => {
+  const tenantId: string = process.env.DVSA_MES_AzureAD_TenantId || '';
+  const applicationId: string = process.env.DVSA_MES_AzureAD_ClientId || '';
+
+  ensureNotNullOrEmpty(tenantId, 'tenantId');
+  ensureNotNullOrEmpty(applicationId, 'applicationId');
+
+  const openidConfig = await (await nodeFetch(
+    `https://login.microsoftonline.com/${tenantId}/.well-known/openid-configuration`,
+  )).json();
+
+  if (openidConfig.error_description) {
```

Comment:
How comes we're not wrapping the `await` in a try/catch? That's the standard convention for doing error handling here

Reply:
Are you referring to the use of `error_description`?  Actual internal errors in the async call will cause an exception to be thrown.

The `error_description` field is populated when the request responds with a "400 bad request".  That could happen if the config has an incorrect tenantId, or if the configuration of that tenant in Azure is invalid, say its subscription has expired.  If we weren't to check the `error_description` field, then we would instead just get an obscure, hard to debug error thrown later from the `jwksClient` because of passing an undefined `jwks_uri`.